File: typescript/problem_85.ts
export function selectByBit(x: number, y: number, b: number): number {
    return x * b + y * (1 - b);
}

export function problem85(): void {
    console.log("Problem 85:");
    if (selectByBit(10,20,1) !== 10) throw new Error("assertion failed");
    if (selectByBit(10,20,0) !== 20) throw new Error("assertion failed");
    if (selectByBit(5,7,1)   !== 5)  throw new Error("assertion failed");
    console.log("All Tests Passed");
}

File: lua/problem_85.lua
local M = {}

function M.select_by_bit(x, y, b)
    return x * b + y * (1 - b)
end

function M.problem85()
    print("Problem 85:")
    assert(M.select_by_bit(10, 20, 1) == 10)
    assert(M.select_by_bit(10, 20, 0) == 20)
    assert(M.select_by_bit(5,  7,  1) == 5)
    print("All Tests Passed")
end

return M
